Redis config

mtkat4k opened the pull request.

mtkat4k committed:
reflow and refactor code

anshul-fk commented on lib/sqspoller/sqs_poller.rb:
```diff
@@ -42,34 +46,42 @@ def start_poller_with_config(config, queue_config_name, access_key_id, secret_ac
+                                     logger_file
+            qc.start
+            qc.threads.each do |thread|
+              thread.join
```
thread.join should be done out of queues_config.keys loop.

anshul-fk commented on lib/sqspoller/sqs_poller.rb:
```diff
@@ -42,70 +48,149 @@ def daemonize(filename)
+                      }
+
+        if redis_or_file
+          if REDIS.get('random_key') == nil
```
Not sure if this is valid REDIS connectivity check.
If it is not able to connect to REDIS, I think it will timeout and throw a exception.

anshul-fk commented on lib/sqspoller/sqs_poller.rb:
```diff
@@ -42,70 +48,149 @@ def daemonize(filename)
+      def start_queues_with_config poller_args
+        @logger = Logger.new(poller_args[:log_filename])
+        poller_args[:logger] = @logger
+        @logger.info "Get config"
```
this line can be removed.

anshul-fk commented on lib/sqspoller/sqs_poller.rb:
```diff
@@ -42,70 +48,149 @@ def daemonize(filename)
+      end
+
+      def start_all_queues_with_refresh queues_config, message_delegator, poller_args, refresh_interval
+        @logger.info "Start all queues with refresh"
```
remove this line.

anshul-fk commented on lib/sqspoller/sqs_poller.rb:
```diff
@@ -42,70 +48,149 @@ def daemonize(filename)
+          queue_names = queues_config.keys
+          @logger.info "  Going to start queues: #{queue_names.join ', '}"
+          queues_config.keys.each do |queue|
+            @logger.info "    Checking queue #{queue}"
```
remove this line.

anshul-fk commented on lib/sqspoller/sqs_poller.rb:
```diff
@@ -42,70 +48,149 @@ def daemonize(filename)
+        loop do
+          Thread.pass
+          queue_names = queues_config.keys
+          @logger.info "  Going to start queues: #{queue_names.join ', '}"
```
remove this line.

anshul-fk commented on lib/sqspoller/sqs_poller.rb:
```diff
@@ -42,70 +48,149 @@ def daemonize(filename)
+        @logger.info "Start all queues with refresh"
+        queues = {}
+        loop do
+          Thread.pass
```
I don't think Thread.pass is needed.

mtkat4k committed:
PLE-166 SqsPoller changes
create SnsForwarder worker
pull config from Redis with refresh interval
check maintenance window
long poll sqs
